lib/fluent/plugin/auditd.rb

```diff
@@ -0,0 +1,123 @@
+module Fluent
+  class Auditd
+    class AuditdParserException < StandardError
+    end
+    
+    MSG = "msg"
+    TIMESTAMP = "time"
+    AUDIT = "audit"
+
+    def parse_auditd_line(line)
+      states = {}
+      states[:start] = 0
+      states[:key] = 1
+      states[:value] = 2
+      state = states[:start]
+      stack = ["$"]
+      line << "$"
+      nested = false
+      key, nested_key, value = nil
+      result = {}
+      for i in 0...line.length do
+        if state == states[:start]
```

Comment:
when statements might make this big if/else more legible: https://stackoverflow.com/questions/948135/how-to-write-a-switch-statement-in-ruby